sdk/core/Azure.Core/tests/TestFramework/TestRecording.cs

```diff
@@ -210,14 +210,16 @@ public void DisableIdReuse()
 
         private class TestCredential : TokenCredential
         {
-            public override ValueTask<string> GetTokenAsync(string[] scopes, CancellationToken cancellationToken)
+            public override async Task<AccessToken> GetTokenAsync(string[] scopes, CancellationToken cancellationToken)
             {
-                return new ValueTask<string>(GetToken(scopes, cancellationToken));
+                await Task.CompletedTask;
```

Comment:
Perhaps I'm missing something important, but should we consider using `Task.FromResult` here, rather than awaiting on the completed task?